web-registry/src/lib/asyncSome.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * This function takes an array and an async function
+ * you wish to apply to each array index until one index
+ * returns true. It will then short-circuit, like Array.some.
+ * This is useful when you want to avoid making extra
+ * unnecessary async requests.
+ */
+export const asyncSome = async (
+  arr: any[],
```

Comment:
Much better, thanks.

Reply:
would it be able to use type inference at this call site since these classes are typed? https://github.com/regen-network/regen-web/pull/1134/files#diff-e0ee8396192e767af18980e011b9548901f278705183635d097afbeff2cea60cR24

learned something new here :-)